org.getmonero.Wallet.json

```diff
@@ -0,0 +1,58 @@
+{
+    "app-id": "org.getmonero.Wallet",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "1.6",
+    "branch": "stable",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "monero-wallet-gui",
+    "finish-args": [
+        /* X11 + XShm access */
+        "--share=ipc", "--socket=x11",
+
+        /* Needs to talk to the network: */
+        "--share=network",
+
+        /* Needed for dconf to work */
+        "--filesystem=home:rw"
+
+    ],
+    "cleanup": ["/include", "/lib/pkgconfig",
+                "/share/pkgconfig", "/share/aclocal",
+                "/man", "/share/man", 
+                "*.la", "*.a"],
+    "modules": [
+        {
+            "name": "monero-core-gui",
```

Comment:
Why not build it from source, given that it's open source?

Reply:
That was my first intention (and still something I'm looking in to), but for some reason there is some QT-related segfaults when I run it in a flatpak. Not sure why, so for now I just settled with repackaging. But there are in fact a benefit of running a repackaged version, and that is that it is actually possible to verify that it is the exact same binary as released on getmonero.org. And since it is a wallet, it is important that no malicious code is involved. I have thought about creating a simple script that would download the official binary, and do sha comparison of the binary files in the flatpak and the officially released tar.gz file. Just so the users can feel safe.

Reply:
It'd be ideal to support the other architectures on Flathub if possible - one of the big advantages Flathub can provide for app developers is ARM and ARM64 packaging.

Reply:
I think you have that backwards, you can verify no malicious code is involved if you build it but downloading binaries you cannot. So I think the security argument is just wrong.

Adding to that the point @nedrichards made I believe there are only benefits to building it so we should look into those errors you are having.

Reply:
Well, the trust is in the source as you say. But the users here are using binaries, so they have to trust flathub.org or getmonero.org. What I'm saysing is that Monero Core is already trusted by most users, hence it might be a point in providing their binaries. 
Anyway, as I said, my original plan was to build from source. This re-wrap is just a way to get a flatpak out the door until I can figure out why things are not working. 

Here is my repository for the source build:
https://github.com/snaggen/monero-wallet-flatpak/

Unfortunately I seem to have lost the latest work on that, where I used the 5.9 version of the KDE runtime, and built xmllistmodel from source also (since the one packaged there was built against another version of the KDE runtime, so it is no good). However, the xmllistmodel is a little tricky to build from source, or rather I used a "simple" build system to be able to manually copy the files to the correct direectory, since using the cmake will end up trying to install the libs on the read only file system. 
If someone feels comfortable with debuging QT and KDE I can see if I can recreate my previous work, so you can continue where I got stuck. 

Anyway, I think the re-wrapped version would be nice to have in the mean time.